src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -243,18 +244,23 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
     register_cli_argument('network nic {}'.format(item), 'load_balancer_inbound_nat_rule_ids', options_list=('--lb-inbound-nat-rules',), nargs='+', validator=validate_inbound_nat_rule_id_list, help='Space separated list of names or IDs of load balancer inbound NAT rules to associate with the NIC. If names are used, --lb-name must be specified.', completer=get_lb_subresource_completion_list('inboundNatRules'))
 
 # NIC ScaleSet
-register_cli_argument('network nic scale-set', 'virtual_machine_scale_set_name', options_list=('--vm-scale-set',), completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
+register_cli_argument('network nic scale-set', 'virtual_machine_scale_set_name', options_list=('--vm-scale-set',), completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part=None)
 register_cli_argument('network nic scale-set', 'virtualmachine_index', options_list=('--vm-index',))
-register_cli_argument('network nic scale-set', 'network_interface_name', id_part='child_name')
+register_cli_argument('network nic scale-set', 'network_interface_name', nic_type, id_part=None)
 
-register_cli_argument('network nic ip-config', 'network_interface_name', options_list=('--nic-name',), metavar='NIC_NAME', help='The network interface (NIC).', id_part='name', completer=get_resource_name_completion_list('Microsoft.Network/networkInterfaces'))
+register_cli_argument('network nic scale-set list', 'virtual_machine_scale_set_name', options_list=('--vm-scale-set',), completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachineScaleSets'), id_part='name')
```

Comment:
There is id for each instance say `/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/YUGANGWVMSS/providers/Microsoft.Compute/virtualMachineScaleSets/yugangwvmss-1/virtualMachines/0`
But at the same time, this command will be moved to under `az vmss`, let us hold on to it